### PR TITLE
[SYCL][Test E2E] Replace 'double' type with 'float' type

### PR DESCRIPTION
**Pull request by myler**
To avoid 'aspect fp64 is not supported on the device' in Gen12+ GPUs.

### CODE REVIEW

File: sycl/test-e2e/Regression/commandlist/Inputs/FindPrimesSYCL.cpp [hunk 1]
```diff
@@ -105,7 +105,7 @@ float find_prime_s(work *w) {
     }
   }
 
-  std::chrono::duration<double> diff{0};
```
Comment (steffenlarsen) [inline sycl/test-e2e/Regression/commandlist/Inputs/FindPrimesSYCL.cpp:108]: I don't think we need this to use float. This is on host so it should be fine.

File: sycl/test-e2e/Regression/commandlist/Inputs/main.cpp [hunk 2]
```diff
@@ -172,9 +172,9 @@ int main(int argc, char *argv[]) {
     avgtime += w.run;
 
 #ifdef DEBUG
-    std::chrono::duration<double> d1 = w.start - minstart;
-    std::chrono::duration<double> d2 = w.stop - minstart;
-    std::chrono::duration<double> d3 = w.stop - w.start;
```
Comment (steffenlarsen) [inline sycl/test-e2e/Regression/commandlist/Inputs/main.cpp:177]: Same as my other comment. I don't think we need to change this file at all.

File: sycl/test-e2e/Regression/commandlist/Inputs/FindPrimesSYCL.cpp (hunk 1 above)
Comment (myler) [reply sycl/test-e2e/Regression/commandlist/Inputs/FindPrimesSYCL.cpp]: Removed.

File: sycl/test-e2e/Regression/commandlist/Inputs/main.cpp (hunk 2 above)
Comment (myler) [reply sycl/test-e2e/Regression/commandlist/Inputs/main.cpp]: Removed the type change on host side.
Verified on Linux + Gen12, test can pass with only following change:
```
-            const int upper_bound = sycl::sqrt(1.0 * number) + 1;
+            const int upper_bound = sycl::sqrt(1.0f * number) + 1;
```

**Review by steffenlarsen — APPROVED** [review]
LGTM!

**Comment by myler** [comment]
one test is reached timeout limit on Gen12:
```
Timed Out Tests (1):
  SYCL :: Basic/stream/auto_flush.cpp
```

**Comment by myler** [comment]
Ping @intel/llvm-gatekeepers this PR is ready to merge.

**Comment by bader** [comment]
@myler, please, resolve merge conflicts.

**Comment by bader** [comment]
Superseded by https://github.com/intel/llvm/pull/10616.

**Pull request closed without merging**